src/Orleans.Core/Core/ClientBuilderExtensions.cs

```diff
@@ -188,7 +189,28 @@ public static IClientBuilder ConfigureLogging(this IClientBuilder builder, Actio
             string serviceId = ClusterOptions.DevelopmentServiceId,
             string clusterId = ClusterOptions.DevelopmentClusterId)
         {
-            return builder.UseStaticClustering(gatewayPorts.Select(p => new IPEndPoint(IPAddress.Loopback, p)).ToArray())
+            return builder.UseLocalhostClustering(gatewayPorts.AsSpan(), serviceId, clusterId);
+        }
+
+        /// <summary>
+        /// Configures the client to connect to a silo on the localhost.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="gatewayPorts">The local silo gateway ports.</param>
+        /// <param name="serviceId">The service id.</param>
+        /// <param name="clusterId">The cluster id.</param>
+        public static IClientBuilder UseLocalhostClustering(this IClientBuilder builder,
```

Comment:
I feel this change is unnecessary, but the rest look good to me. Generally speaking, APIs which are only called once are not candidates for spanification

Reply:
Done